connectivity/tests/perfpod.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/defaults"
+)
+
+const (
+	messageSize = 1024
+)
+
+// Network Performance
+func NetperfPodtoPod(n string) check.Scenario {
+	return &netPerfPodtoPod{
+		name: n,
+	}
+}
+
+type netPerfPodtoPod struct {
+	name string
+}
+
+func (s *netPerfPodtoPod) Name() string {
+	tn := "perf-pod-to-pod"
+	if s.name == "" {
+		return tn
+	}
+	return fmt.Sprintf("%s:%s", tn, s.name)
+}
+
+func (s *netPerfPodtoPod) Run(ctx context.Context, t *check.Test) {
+	for _, c := range t.Context().PerfClientPods() {
+		c := c
+		for _, server := range t.Context().PerfServerPod() {
+			action := t.NewAction(s, "iperf-tcp", &c, server)
+			action.CollectFlows = false
+			action.Run(func(a *check.Action) {
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_RR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_CRR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_STREAM", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "UDP_RR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "UDP_STREAM", a, t.Context().PerfResults, 1)
+			})
+		}
+	}
+}
+
+func netperf(ctx context.Context, sip string, podname string, test string, a *check.Action, result map[check.PerfTests]check.PerfResult, samples int) {
+	// Define test about to be executed and from which pod
+	k := check.PerfTests{
+		Pod:  podname,
+		Test: test,
+	}
+
+	// Allow the user to override aspects of the test
+	iteration, _ := strconv.Atoi(os.Getenv("samples"))
+	duration, _ := strconv.Atoi(os.Getenv("duration"))
```

Comment:
Where are these env vars coming from? Would be good to document that.

Reply:
Removing these... In previous commits I was using them, but with the latest change to have the performance tests exist on their own, I will be removing these 👍🏽